modules/new_serverpod_auth/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,589 @@
+      userProfile = userProfile.copyWith(
+        userName: modifiedProfile.userName,
+        fullName: modifiedProfile.fullName,
+        email: modifiedProfile.email?.toLowerCase().trim(),
+      );
+    }
+
+    userProfile = await UserProfile.db.updateRow(
+      session,
+      userProfile,
+    );
+
+    await _invalidateCacheForUser(session, userProfile.authUserId);
+
+    await UserProfileConfig.current.onAfterUserProfileUpdated?.call(
+      session,
+      userProfile.toModel(),
+      transaction: transaction,
+    );
+
+    return userProfile;
+  }
+
+  static String _userProfileCacheKey(
+    /// ID of then [AuthUser]
+    final UuidValue authUserId,
+  ) {
+    return 'serverpod_auth_profile_$authUserId';
+  }
+
+  /// Invalidates the cache for a user and makes sure the next time a user profile
+  /// is fetched it's fresh from the database.
+  static Future<void> _invalidateCacheForUser(
+    final Session session,
+    final UuidValue authUserId,
+  ) async {
+    final cacheKey = _userProfileCacheKey(authUserId);
+
+    await session.caches.local.invalidateKey(cacheKey);
+  }
+
+  static Future<UserProfile?> _maybeFindUserProfile(
+    final Session session,
+    final UuidValue authUserId, {
+    required final Transaction? transaction,
+  }) async {
+    final userProfile = await UserProfile.db.findFirstRow(
+      session,
+      where: (final t) => t.authUserId.equals(authUserId),
+      include: UserProfile.include(
+        image: UserProfileImage.include(),
+      ),
+      transaction: transaction,
+    );
+
+    return userProfile;
+  }
+
+  static Future<UserProfile> _findUserProfile(
+    final Session session,
+    final UuidValue authUserId, {
+    required final Transaction transaction,
+  }) async {
+    final userProfile = await _maybeFindUserProfile(
+      session,
+      authUserId,
+      transaction: transaction,
+    );
+
+    if (userProfile == null) {
+      throw UserProfileNotFoundException(authUserId);
+    }
+
+    return userProfile;
+  }
+
+  static Future<Uint8List> _encodeImage(final Image image) {
+    return Isolate.run(
+      () => switch (UserProfileConfig.current.userImageFormat) {
+        UserProfileImageType.jpg => encodeJpg(
+            image,
+            quality: UserProfileConfig.current.userImageQuality,
+          ),
+        UserProfileImageType.png => encodePng(image),
+      },
+    );
+  }
+}
+
+extension on Session {
+  /// Runs the closure [f] in isolation, inside the parent [Transaction] (if any).
+  ///
+  /// Either creates a new transaction for [f],
+  /// or creates a savepoint inside the given transaction,
+  /// and would discard any modifications if [f] fails.
+  Future<R> transactionOrSavepoint<R>(
+    final TransactionFunction<R> f, {
+    required Transaction? transaction,
+  }) async {
+    // Use the implicit transaction from tests
+    // ignore: invalid_use_of_visible_for_testing_member
+    transaction ??= this.transaction;
```

Comment:
I suppose this is fine here, as the low-level DB helpers would just do this as well.

So for the test code there is never any need to use the `transaction: ` parameter explicitly.

Reply:
This has been removed as requested.